### PR TITLE
Update elm-arithmetic to 0.19 and add tests

### PR DESCRIPTION
**Pull request by JoshuaHall**
- Updated to 0.19 (elm-package.json is now elm.json, infix operators are gone, variables can no longer be shadowed, % operator is now modBy etc.)
- removed `squareRoot`, as `sqrt` is in `elm/core` now
- Added tests*** using https://github.com/elm-explorations/test/
- minor elm-format changes

***One test is still not passing for the `extendedGcd` function, and I'm unsure if it's an issue in my test or the library for now. I thought I'd put this pull request out though so I can maybe get some help in looking over the code.

### CODE REVIEW

File: src/Arithmetic.elm [hunk 1]
```diff
@@ -221,33 +247,58 @@ intCubeRoot =
...
     let
         r =
-            n % 63
+            modBy 63 n
     in
-        (r == 0 || r == 1 || r == 8 || r == 27 || r == 28 || r == 35 ||
-         r == 36 || r == 55 || r == 62) && intCubeRoot n ^ 3 == n
+    (r
+        == 0
+        || r
```
Comment (Janiczek) [inline src/Arithmetic.elm]: ouch, some parentheses would help (elm-format respects those)
```elm
((r == 0)
  || (r == 1)
  || ...
)
```

Comment (JoshuaHall) [reply src/Arithmetic.elm]: True, I'm not sure how I didn't see that! That is indeed hard to read lol

Comment (JoshuaHall) [reply src/Arithmetic.elm]: I pushed a fix, it's not my favorite syntax but it should be easier to read.

**Comment by JoshuaHall** [comment]
The specific test that is failing is this one:
```
, fuzz2 int int "extendedGcd fuzzy test 2" <|
    \a b ->
        let
            ( d, u, v ) =
                extendedGcd a b
        in
        ((a * u) + (b * v) == d && gcd a b == d)
            |> Expect.true "should follow the au + bv = d and gcd a b = d identity"
```
It's just plugging the numbers back in and checking that they make sense. It seems to work for small numbers, but when it gets to really large numbers, it starts to fail. Here are some sample inputs that caused a failure:

```
extendedGcd 88352448 223957979
extendedGcd 8011999 2320846895
extendedGcd 7351762 2552966159
```

**Comment by JoshuaHall** [comment]
I fixed the failing test. It was because of a lack of integer bounds checking, and when the numbers got out of the JavaScript int range things got weird. For now I decided to put an `Expect.pass` there, since I don't think this library should be concerned with whether its functions work for integers that are outside of the supported range 😃

**Comment by lynn** [comment]
Thanks for keeping this library up-to-date! And the tests are a very welcome bonus. You're the best!

**Pull request merged**